api/server/server.go

```diff
@@ -1123,6 +1126,142 @@ func postContainersCopy(eng *engine.Engine, version version.Version, w http.Resp
 	return nil
 }
 
+// handleArchiveError is a convenience function for the following Copy
+// handlers which makes a JSON error message but leaves writing the
+// header and response to the package's common error handler.
+func handleArchiveError(w http.ResponseWriter, err error) error {
```

Comment:
The problem is, now a lower-level routine is transport aware and I'd prefer for it to be transport agnostic.  At some point I think we'll get docker to where the daemon isn't called via http and in those cases setting http headers doesn't make sense.  What if, instead, you simple had your code return JSONError's - then httpError() can translate those.  While JSONError's aren't used much right now I'd like to morph them into the DError stuff I mentioned in my other PR.